File: ai-code-frontend/src/services/backend/friendController.ts
import request from '@/utils/request';
import type { PageResponseVO } from '@/api/vo';

export interface FriendRequestParams {
  addresseeId: string;
  message?: string;
}

export interface FriendRequestListParams {
  status: string;
  pageNum: number;
  pageSize: number;
}

export interface FriendListParams {
  pageNum: number;
  pageSize: number;
}

export interface FriendVO {
  id: number;
  userId: number;
  friendId: number;
  friendName?: string;
  friendAvatar?: string;
  remark?: string;
  status: number;
  createTime: string;
}

export interface FriendRequestVO {
  id: number;
  requesterId: number;
  requesterName?: string;
  requesterAvatar?: string;
  addresseeId: number;
  message?: string;
  status: number;
  createTime: string;
}

/**
 * 发送好友请求
 */
export async function sendFriendRequest(data: FriendRequestParams) {
  return request.post<FriendRequestParams, PageResponseVO<boolean>>('/friend/request', data);
}

/**
 * 接受好友请求
 */
export async function acceptFriendRequest(requestId: string) {
  return request.put<string, PageResponseVO<boolean>>(`/friend/request/${requestId}/accept`);
}

/**
 * 拒绝好友请求
 */
export async function rejectFriendRequest(requestId: string) {
  return request.put<string, PageResponseVO<boolean>>(`/friend/request/${requestId}/reject`);
}

/**
 * 获取好友请求列表
 */
export async function getFriendRequestList(params: FriendRequestListParams) {
  return request.get<FriendRequestListParams, PageResponseVO<FriendRequestVO[]>>('/friend/requests', { params });
}

/**
 * 获取好友列表
 */
export async function getFriendList(params: FriendListParams) {
  return request.get<FriendListParams, PageResponseVO<FriendVO[]>>('/friend/list', { params });
}

/**
 * 删除好友
 */
export async function deleteFriend(friendId: string) {
  return request.delete<string, PageResponseVO<boolean>>(`/friend/${friendId}`);
}

/**
 * 获取好友详情
 */
export async function getFriendDetail(friendId: string) {
  return request.get<string, PageResponseVO<FriendVO>>(`/friend/${friendId}`);
}

/**
 * 更新好友备注
 */
export async function updateFriendRemark(friendId: string, remark: string) {
  return request.put<{ friendId: string; remark: string }, PageResponseVO<boolean>>('/friend/remark', { friendId, remark });
}

File: src/main/java/com/mashang/aicode/web/controller/FriendController.java
package com.mashang.aicode.web.controller;

import com.baomidou.mybatisplus.extension.plugins.pagination.Page;
import com.mashang.aicode.web.annotation.AuthCheck;
import com.mashang.aicode.web.common.BaseResponse;
import com.mashang.aicode.web.common.PageRequest;
import com.mashang.aicode.web.common.ResultUtils;
import com.mashang.aicode.web.constant.UserConstant;
import com.mashang.aicode.web.exception.BusinessException;
import com.mashang.aicode.web.exception.ErrorCode;
import com.mashang.aicode.web.model.dto.friend.FriendListQueryDTO;
import com.mashang.aicode.web.model.dto.friend.FriendRequestSendDTO;
import com.mashang.aicode.web.model.entity.FriendRequest;
import com.mashang.aicode.web.model.vo.FriendRequestVO;
import com.mashang.aicode.web.model.vo.UserVO;
import com.mashang.aicode.web.service.FriendService;
import com.mashang.aicode.web.service.UserService;
import jakarta.annotation.Resource;
import jakarta.servlet.http.HttpServletRequest;
import lombok.extern.slf4j.Slf4j;
import org.springframework.web.bind.annotation.*;

/**
 * 好友控制器
 */
@RestController
@RequestMapping("/friend")
@Slf4j
public class FriendController {

    @Resource
    private FriendService friendService;

    @Resource
    private UserService userService;

    /**
     * 发送好友请求
     *
     * @param requestDTO 发送好友请求DTO
     * @param request    HTTP请求
     * @return 好友请求
     */
    @PostMapping("/request")
    @AuthCheck(mustRole = UserConstant.DEFAULT_ROLE)
    public BaseResponse<FriendRequest> sendFriendRequest(@RequestBody FriendRequestSendDTO requestDTO, HttpServletRequest request) {
        Long userId = userService.getLoginUser(request).getId();
        FriendRequest friendRequest = friendService.sendFriendRequest(requestDTO, userId);
        return ResultUtils.success(friendRequest);
    }

    /**
     * 接受好友请求
     *
     * @param requestId 请求ID
     * @param request   HTTP请求
     * @return 是否成功
     */
    @PutMapping("/request/{requestId}/accept")
    @AuthCheck(mustRole = UserConstant.DEFAULT_ROLE)
    public BaseResponse<Boolean> acceptFriendRequest(@PathVariable Long requestId, HttpServletRequest request) {
        Long userId = userService.getLoginUser(request).getId();
        boolean success = friendService.acceptFriendRequest(requestId, userId);
        return ResultUtils.success(success);
    }

    /**
     * 拒绝好友请求
     *
     * @param requestId 请求ID
     * @param request   HTTP请求
     * @return 是否成功
     */
    @PutMapping("/request/{requestId}/reject")
    @AuthCheck(mustRole = UserConstant.DEFAULT_ROLE)
    public BaseResponse<Boolean> rejectFriendRequest(@PathVariable Long requestId, HttpServletRequest request) {
        Long userId = userService.getLoginUser(request).getId();
        boolean success = friendService.rejectFriendRequest(requestId, userId);
        return ResultUtils.success(success);
    }

    /**
     * 获取收到的好友请求
     *
     * @param status    请求状态
     * @param pageNum   页码
     * @param pageSize  每页大小
     * @param request   HTTP请求
     * @return 好友请求分页列表
     */
    @GetMapping("/requests")
    @AuthCheck(mustRole = UserConstant.DEFAULT_ROLE)
    public BaseResponse<Page<FriendRequestVO>> getReceivedFriendRequests(
            @RequestParam(value = "status", defaultValue = "PENDING") String status,
            @RequestParam(value = "pageNum", defaultValue = "1") Integer pageNum,
            @RequestParam(value = "pageSize", defaultValue = "10") Integer pageSize,
            HttpServletRequest request) {
        Long userId = userService.getLoginUser(request).getId();
        Page<FriendRequestVO> page = friendService.getReceivedFriendRequests(status, pageNum, pageSize, userId);
        return ResultUtils.success(page);
    }

    /**
     * 获取好友列表
     *
     * @param pageNum   页码
     * @param pageSize  每页大小
     * @param request  HTTP请求
     * @return 好友用户分页列表
     */
    @GetMapping("/list")
    @AuthCheck(mustRole = UserConstant.DEFAULT_ROLE)
    public BaseResponse<Page<UserVO>> getFriendList(
            @RequestParam(value = "pageNum", defaultValue = "1") Integer pageNum,
            @RequestParam(value = "pageSize", defaultValue = "20") Integer pageSize,
            HttpServletRequest request) {
        Long userId = userService.getLoginUser(request).getId();
        FriendListQueryDTO queryDTO = new FriendListQueryDTO();
        queryDTO.setPageNum(pageNum);
        queryDTO.setPageSize(pageSize);
        Page<UserVO> page = friendService.getFriendList(queryDTO, userId);
        return ResultUtils.success(page);
    }

    /**
     * 删除好友
     *
     * @param friendId 好友ID
     * @param request  HTTP请求
     * @return 是否成功
     */
    @DeleteMapping("/{friendId}")
    @AuthCheck(mustRole = UserConstant.DEFAULT_ROLE)
    public BaseResponse<Boolean> deleteFriend(@PathVariable Long friendId, HttpServletRequest request) {
        Long userId = userService.getLoginUser(request).getId();
        boolean success = friendService.deleteFriend(friendId, userId);
        return ResultUtils.success(success);
    }
}